main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"os"
+	glog "github.com/Sirupsen/logrus"
+	"github.com/projectcalico/k8s-policy/pkg/controllers/namespace"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+func main() {
+
+	logLevel, err := glog.ParseLevel(os.Getenv("LOG_LEVEL"))
+	if(err!=nil){
+		// Defaulting log level to INFO
+		logLevel = glog.InfoLevel
+	}
+
+	glog.SetLevel(logLevel)
+
+	k8sClientset, calicoClient, err := getClients()
+
+	if err != nil {
+		glog.Fatal(err)
+	}
+	controller := namespace.NewNamespaceController(k8sClientset, calicoClient)
+
+	stop := make(chan struct{})
+	defer close(stop)
+
+	reconcilerPeriod, exists := os.LookupEnv("RECONCILER_PERIOD")
+	if(!exists){
+		// Defaulting to 5 mins
+		reconcilerPeriod = "5m"
+	} 
+	
+	go controller.Run(5, reconcilerPeriod, stop)
+
+	// Wait forever.
+	select {}
+}
+
+// Fuction that returns kubernetes and calico clients.
+// Function collects connection infromation from environment variables
+// as well as kubeconfig file. Environment variables override configs
+// provided in kubeconfig file.
+func getClients() (*kubernetes.Clientset, *client.Client, error) {
+
+	cconfig, err := client.LoadClientConfig("")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get Calico client
+	calicoClient, err := client.New(*cconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	k8sConfig := cconfig.Spec.KubeConfig
```

Comment:
I don't think this is actually right.

The Calico client is going to be an etcd client, which means the `config.Spec.KubeConfig` is going to be empty.